telemetry/src/lib.rs

```diff
@@ -2,13 +2,39 @@ use std::io::IsTerminal as _;
 
 use tracing::level_filters::LevelFilter;
 use tracing_subscriber::{
-    layer::SubscriberExt as _, util::SubscriberInitExt as _, EnvFilter,
+    layer::SubscriberExt as _,
+    EnvFilter,
 };
 
+use opentelemetry::{global, KeyValue};
+use opentelemetry::trace::TracerProvider;
+use opentelemetry_sdk::propagation::TraceContextPropagator;
+use opentelemetry_otlp::WithExportConfig;
+use opentelemetry_sdk::{
+    trace::{self, Sampler},
+    runtime::Tokio,
+    Resource,
+};
+use tracing_subscriber::util::SubscriberInitExt;
+
+/// A struct controlling how telemetry will be configured (logging + optional OpenTelemetry).
 #[derive(Debug)]
 pub struct TelemetryConfig {
     syslog_identifier: Option<String>,
     global_filter: EnvFilter,
+
+    /// If true, enable OTLP tracing via OpenTelemetry.
+    use_otel: bool,
+
+    /// The service name used in opentelemetry's `Resource`.
```

Comment:
all the open telemetry settings should live in one OpenTelemetryConfig struct, with a single `otel: Option<OpenTelemetryConfig>` in the `TelemetryConfig` struct.